cpu/msp430fxyz/include/periph_cpu.h

```diff
@@ -26,7 +26,69 @@
 extern "C" {
 #endif
 
-/* more to come here... */
+/**
+ * @brief   Define a custom type for GPIO pins
+ * @{
+ */
+#define HAVE_GPIO_T
+typedef uint16_t gpio_t;
+/** @} */
+
+/**
+ * @brief   Definition of a fitting UNDEF value
+ */
+#define GPIO_UNDEF          (0xffff)
+
+/**
+ * @brief   Mandatory function for defining a GPIO pins
+ * @{
+ */
+#define GPIO(x, y)          ((gpio_t)(((x & 0xff) << 8) | (1 << (y & 0xff))))
+
+/**
+ * @brief   Override direction values
+ * @{
+ */
+#define HAVE_GPIO_DIR_T
+typedef enum {
+    GPIO_DIR_IN =  0x00,    /**< configure pin as input */
+    GPIO_DIR_OUT = 0xff,    /**< configure pin as output */
+} gpio_dir_t;
+/** @} */
+
+/**
+ * @brief   Override flank selection values
+ * @{
+ */
+#define HAVE_GPIO_FLANK_T
+typedef enum {
+    GPIO_FALLING = 0xff,        /**< emit interrupt on falling flank */
+    GPIO_RISING  = 0x00,        /**< emit interrupt on rising flank */
+    GPIO_BOTH    = 0xab         /**< not supported -> random value*/
+} gpio_flank_t;
+/** @} */
+
+/**
+ * @brief   Available ports on MSP430 platforms
+ */
+enum {
+    P0 = 0,                 /**< PORT 0 */
+    P1 = 1,                 /**< PORT 1 */
+    P2 = 2,                 /**< PORT 2 */
+    P3 = 3,                 /**< PORT 3 */
+    P4 = 4,                 /**< PORT 4 */
+    P5 = 5,                 /**< PORT 5 */
+    P6 = 6,                 /**< PORT 6 */
+};
```

Comment:
not yet, but they will be. These enum you basically use, when you configure devices on your board - e.g. you say something like `#define CC110X_CS      GPIO(P0,3)`